trin-history/src/validation.rs

```diff
@@ -191,8 +275,149 @@ mod tests {
             block_hash: header.hash().0,
         });
         chain_history_validator
-            .validate_content(&content_key, &header_bytelist)
+            .validate_content(&content_key, &header_bytelist.to_vec())
+            .await
+            .unwrap();
+    }
+
+    #[tokio::test]
+    async fn validate_block_body() {
+        let server = setup_mock_infura_server();
+
+        let ssz_block_body = hex::decode(
+            std::fs::read_to_string("../trin-core/src/types/assets/block_body_14764013.hex")
+                .unwrap(),
+        )
+        .unwrap();
+        let block_body_bytelist: VariableList<_, typenum::U16384> =
+            VariableList::from(ssz_block_body);
+
+        let infura_url = server.url("/14764013");
+        let header_oracle = Arc::new(RwLock::new(HeaderOracle {
+            infura_url,
+            ..HeaderOracle::default()
+        }));
+        let mut chain_history_validator = ChainHistoryValidator { header_oracle };
+        let content_key = block_14764013_body_key();
+
+        chain_history_validator
+            .validate_content(&content_key, &block_body_bytelist.to_vec())
             .await
             .unwrap();
     }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn invalidate_block_body() {
```

Comment:
Probably want to simulate both kinds of failures in independent tests: a tx root failure and an uncle hash failure.